Specs/Scene/ImageryLayerCollectionSpec.js

```diff
@@ -258,6 +258,260 @@ describe(
+        scene.destroyForSpecs();
+      });
+
+      beforeEach(function () {
+        globe.imageryLayers.removeAll();
+      });
+
+      /**
+       * Repeatedly calls update until the load queue is empty.  Returns a promise that resolves
+       * once the load queue is empty.
+       */
+      function updateUntilDone(globe) {
+        // update until the load queue is empty.
+        return pollToPromise(function () {
+          globe._surface._debug.enableDebugOutput = true;
+          scene.render();
+          return (
+            globe._surface.tileProvider.ready &&
+            globe._surface._tileLoadQueueHigh.length === 0 &&
+            globe._surface._tileLoadQueueMedium.length === 0 &&
+            globe._surface._tileLoadQueueLow.length === 0 &&
+            globe._surface._debug.tilesWaitingForChildren === 0
+          );
+        });
+      }
+
+      it("returns undefined when pick ray does not intersect surface", function () {
+        var ellipsoid = Ellipsoid.WGS84;
+        camera.lookAt(
+          new Cartesian3(ellipsoid.maximumRadius, 0.0, 0.0),
+          new Cartesian3(0.0, 0.0, 100.0)
+        );
+
+        var ray = new Ray(
+          camera.position,
+          Cartesian3.negate(camera.direction, new Cartesian3())
+        );
+        var imagery = scene.imageryLayers.pickImageryLayers(ray, scene);
+        expect(imagery).toBeUndefined();
+      });
+
+      it("returns undefined when globe has no pickable layers", function () {
+        var ellipsoid = Ellipsoid.WGS84;
+        camera.lookAt(
+          new Cartesian3(ellipsoid.maximumRadius, 0.0, 0.0),
+          new Cartesian3(0.0, 0.0, 100.0)
+        );
+
+        var ray = new Ray(camera.position, camera.direction);
+        var imagery = scene.imageryLayers.pickImageryLayers(ray, scene);
+        expect(imagery).toBeUndefined();
+      });
+
+      it("returns undefined when ImageryProvider does not implement pickFeatures", function () {
+        var provider = {
+          ready: true,
+          rectangle: Rectangle.MAX_VALUE,
+          tileWidth: 256,
+          tileHeight: 256,
+          maximumLevel: 0,
+          minimumLevel: 0,
+          tilingScheme: new GeographicTilingScheme(),
+          errorEvent: new Event(),
+          hasAlphaChannel: true,
+
+          requestImage: function (x, y, level) {
+            return ImageryProvider.loadImage(this, "Data/Images/Blue.png");
+          },
+        };
+
+        globe.imageryLayers.addImageryProvider(provider);
+
+        return updateUntilDone(globe).then(function () {
+          var ellipsoid = Ellipsoid.WGS84;
+          camera.lookAt(
+            new Cartesian3(ellipsoid.maximumRadius, 0.0, 0.0),
+            new Cartesian3(0.0, 0.0, 100.0)
+          );
+
+          var ray = new Ray(camera.position, camera.direction);
+          var imagery = scene.imageryLayers.pickImageryLayers(ray, scene);
+          expect(imagery).toBeUndefined();
+        });
+      });
```

Comment:
This test should be removed because `pickFeatures` isn't really relevant to `pickImageryLayers`

CC https://github.com/CesiumGS/cesium/pull/9651#discussion_r667205720

Reply:
Same comment for the test below. It can be removed.

Reply:
For some reason these tests fail when run individually but pass when run with all `ImageryLayerCollection` tests. I would have expected them to fail in both cases. Could you check why this is happening? It's possible that previous tests are altering the scene in some way so that it always passes.

Passes: http://localhost:8080/Specs/SpecRunner.html?spec=Scene%2FImageryLayerCollection

Fails: http://localhost:8080/Specs/SpecRunner.html?spec=Scene%2FImageryLayerCollection%20pickImageryLayers%20returns%20undefined%20when%20ImageryProvider%20does%20not%20implement%20pickFeatures